### PR TITLE
initialise content length to zero if HEAD request

### PR DESCRIPTION
**Pull request by aidylewis**

### CODE REVIEW

**Comment by Lukasa** [comment]
There we go, builds now! =) Seems we have a linting issue.

**Comment by Lukasa** [comment]
This supersedes #258.

File: test/test_head_request.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class TestHeadRequest(object):
+
+        example_request_headers = [
+            (u':authority', u'example.com'),
+            (u':path', u'/'),
+            (u':scheme', u'https'),
+            (u':method', u'HEAD'),
+            (u':content_length', u'1'),
```
Comment (Lukasa) [inline test/test_head_request.py]: Ah, sorry: this test is not quite right.

Firstly, the `content-length` header should be in the *response*, not the request. Secondly, the header name is `content-length`, not `:content-length`.

**Comment by aidylewis** [comment]
I've squashed the last two commits. This **should** be final https://github.com/python-hyper/hyper-h2/pull/264/commits/98e2cd6fa8fdadf96267b565fbd3b857fe415a92 commit

**Comment by aidylewis** [comment]
Are we now OK to squash?

**Comment by Lukasa** [comment]
Yup, looks good to me! :sparkles:

**Comment by Lukasa** [comment]
Merged. Thank you so much @aidylewis! :sparkles: :cake: :sparkles:

**Comment by aidylewis** [comment]
OK. I am off to lunch now. What next? I was thinking of creating a command line load tool from this library. WDYT?

**Comment by Lukasa** [comment]
I think that's an extremely interesting idea! =) Why not prototype it, see how it looks?

**Pull request merged**